### PR TITLE
 update consensusStatus() interface && split m_signalled of Sealer into m_signalled and m_blockSignalled && add certificate generation scripts && add updatePBFTNode interface

### PR DESCRIPTION
**Pull request by cyjseagull**
1. ConsensusEngineBase.*/PBFTEngine.* : update consensusStatus() interface; (change '.' to '_')
2. Sealer.*/PBFTSealer.* : split m_signalled of Sealer into m_signalled and m_blockSignalled
3. agency.sh/node.sh/sdk.sh : add certificate generation scripts
4. Rpc.*/MinerPrecompile.*: add updatePBFTNode interface

### CODE REVIEW

File: tools/install.sh [hunk 1]
```diff
@@ -188,6 +197,7 @@ Usage()
 Usage:
 Optional:
     -d       Download from GitHub
+    -t       Enable test mode (generate mini-consensus/mini-sync/mini-evm/mini-storage/test_verifier)
     -h       Help
 Example: 
     bash build.sh
```
Comment (bxq2011hust) [inline tools/install.sh]: install.sh?

File: libstorage/MinerPrecompiled.cpp [hunk 2]
```diff
@@ -126,3 +126,21 @@ bytes MinerPrecompiled::call(ExecutiveContext::Ptr context, bytesConstRef param)
 
     return out;
 }
+
+std::string MinerPrecompiled::getNodeID(bytesConstRef param)
+{
+    std::string nodeId = "";
+    try
+    {
+        bytesConstRef data = param.cropped(4);
```
Comment (bxq2011hust) [inline libstorage/MinerPrecompiled.cpp]: The first 4 bytes is nodeID?

File: libstorage/MinerPrecompiled.cpp [hunk 3]
```diff
@@ -126,3 +126,21 @@ bytes MinerPrecompiled::call(ExecutiveContext::Ptr context, bytesConstRef param)
 
     return out;
 }
+
+std::string MinerPrecompiled::getNodeID(bytesConstRef param)
```
Comment (bxq2011hust) [inline libstorage/MinerPrecompiled.cpp]: I think this getNodeID() should not in MinerPrecompiled. Precompiled shouldn't have static function!

File: librpc/Rpc.cpp [hunk 4]
```diff
@@ -846,3 +847,60 @@ std::string Rpc::sendRawTransaction(int _groupID, const std::string& _rlp)
...
+    std::string nodeId = dev::blockverifier::MinerPrecompiled::getNodeID(ref(precompileData));
+    /// check the node Id
+    if (nodeId.size() != 128u)
+    {
+        LOG(ERROR) << "[#isValidNodeId] invalid [nodeId]: " << nodeId << std::endl;
+        return false;
+    }
+    /// the nodeId must be defined in the config.ini
+    bool isValid = false;
+    for (auto miner : ledgerParam->mutableConsensusParam().minerList)
```
Comment (bxq2011hust) [inline librpc/Rpc.cpp]: What if nodeID not in `ledgerParam->mutableConsensusParam().minerList`?

File: libstorage/MinerPrecompiled.cpp (hunk 3 above)
Comment (cyjseagull) [reply libstorage/MinerPrecompiled.cpp]: already delelted

File: librpc/Rpc.cpp (hunk 4 above)
Comment (cyjseagull) [reply librpc/Rpc.cpp]: already delelted

File: tools/install.sh (hunk 1 above)
Comment (cyjseagull) [reply tools/install.sh]: done

**Pull request merged**